### PR TITLE
Feature/jn1 4198

### PR DESCRIPTION
**Pull request by CRMTravis**

### CODE REVIEW

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 1]
```diff
@@ -98,16 +98,25 @@ global with sharing class SkillsBasedRouting {
     //List<String> hashStrings;
     String generatedHash;
 
+    //Todays Date
+    DateTime currentDateTime = System.now();
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:102]: always. set specific static dates, using date.newInstance, currently your unit test becomes unpredictable every day

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/SkillsBasedRouting.cls (hunk 1 above)
Comment (CRMTravis) [reply force-app/main/default/classes/SkillsBasedRouting.cls:102]: I don't understand why would I use static dates if the date is dynamic to whenever the class is instantiated.

Comment (RemarioCRM) [reply force-app/main/default/classes/SkillsBasedRouting.cls:102]: unit-testing is supposed to be predicable , if you use dynamic how will you debug the expectation in the near future

Comment (CRMTravis) [reply force-app/main/default/classes/SkillsBasedRouting.cls:102]: I use the addDays method for variability.

File: force-app/main/default/classes/TestSkillsBasedRouting.cls [hunk 2]
```diff
@@ -87,25 +93,51 @@ public class TestSkillsBasedRouting {
       travisSR.Id,
       annottoBaySkill.Id
     );
+    travisAnnottoBaySkill.EffectiveStartDate = System.now().addDays(-1);
```
Comment (RemarioCRM) [inline force-app/main/default/classes/TestSkillsBasedRouting.cls:96]: modify the test setup to be leaner.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 3]
```diff
@@ -52,7 +55,10 @@ global with sharing class SkillsBasedRouting {
     return pq;
   }
 
-  public static void routeUsingSkillsTier2(List<Lead> leads) {
+  public static void routeUsingSkillsTier2(
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:58]: add jsdocs

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 4]
```diff
@@ -12,10 +12,13 @@ global with sharing class SkillsBasedRouting {
   public static final String TAG = 'SkillsBasedRouting';
   public static Boolean inRecursiveUpdate = false;
 
-  public static void StartRoutingUsingTier2(List<Lead> leads) {
+  public static void StartRoutingUsingTier2(
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:15]: add jsdocs

File: force-app/main/default/classes/SkillsBasedRouting.cls [hunk 5]
```diff
@@ -103,11 +109,17 @@ global with sharing class SkillsBasedRouting {
...
         (
-          SELECT SkillId, Skill.MasterLabel
+          SELECT
+            SkillId,
+            Skill.MasterLabel,
+            EffectiveStartDate,
+            EffectiveEndDate
           FROM ServiceResourceSkills
           WHERE
-            SKill.Id IN :mappedLocationSkills.keySet()
```
Comment (RemarioCRM) [inline force-app/main/default/classes/SkillsBasedRouting.cls:109]: add some inline comments to explain what's going on with this query.

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

File: force-app/main/default/classes/SkillsBasedRouting.cls (hunk 5 above)
Comment (RemarioCRM) [reply force-app/main/default/classes/SkillsBasedRouting.cls:109]: all methods in the skill-based class should have a method description

File: force-app/main/default/classes/TestSkillsBasedRouting.cls [hunk 6]
```diff
@@ -2,367 +2,638 @@
...
   @isTest
-  public static void shouldrouteUsingSkillsTier2Scenario1() {
-    User travis = [SELECT Id FROM User WHERE UserName = 'Travis@test.com'];
+  public static void shouldRouteLeadScenario1() {
+    List<ServiceResourceSkill> serviceResourceSkills = new List<ServiceResourceSkill>();
+    //User to run system
+    User travis = [SELECT Id FROM User WHERE UserName = 'TestTravis@test.com'];
     System.runAs(travis) {
       Group salesTeam = [
         SELECT Id
```
Comment (RemarioCRM) [inline force-app/main/default/classes/TestSkillsBasedRouting.cls:96]: use this method instead Util.getQueueId

File: force-app/main/default/classes/TestSkillsBasedRouting.cls [hunk 7]
```diff
@@ -2,367 +2,638 @@
...
       salesTeamRemario.UserOrGroupId = remario.Id;
       salesTeamRemario.GroupId = salesTeam.Id;
       salesTeamMembers.add(salesTeamRemario);
-
       GroupMember salesTeamAbigail = new GroupMember();
       salesTeamAbigail.UserOrGroupId = abigail.Id;
       salesTeamAbigail.GroupId = salesTeam.Id;
       salesTeamMembers.add(salesTeamAbigail);
-
       insert salesTeamMembers;
```
Comment (RemarioCRM) [inline force-app/main/default/classes/TestSkillsBasedRouting.cls:84]: use CreateGroupMember from testdata factory

**Review by RemarioCRM — CHANGES_REQUESTED** [review]

**Review by RemarioCRM — APPROVED** [review]

**Pull request merged**